### DISCUSSION TITLE
【讨论】不知道为什么layui经常有下面这样不符合w3c规范的css代码

### DISCUSSION THREAD

**Discussion by cvpv** [layui/layui#1487] [讨论]
![image](https://github.com/layui/layui/assets/84720311/3de69489-a83d-4ad3-b4c6-063daabcfb14)

**Comment by cvpv** [comment]
如果要注释就直接移除相关代码。
这样写既冗余又不规范

**Comment by sentsim** [comment]
不知道是正常的，这些都是最早兼容 IE6, IE7 留下的代码，很多年轻人可能连 IE6 是什么都没听过，没有经历过那个时代的折磨，是很难理解各种 hack 方案对前端开发带来的挑战。欢迎提交 PR 改善代码，当前版本的 layer，只需兼容 ie8+ 即可

**Comment by cvpv** [comment]
> 不知道是正常的，这些都是最早兼容 IE6, IE7 留下的代码，很多年轻人可能连 IE6 是什么都没听过，没有经历过那个时代的折磨，是很难理解各种 hack 方案对前端开发带来的挑战。欢迎提交 PR 改善代码，当前版本的 layer，只需兼容 ie8+ 即可

原来是这样，受教了

**Comment by cvpv** [comment]
我去百度了下得到下面的解答：

> *right 并不是CSS中的有效代码，它只是一种hack（即技巧）方式，用于解决早期版本的Internet Explorer（IE6和IE7）中的一些特殊问题。

具体来说，*right 是指在CSS样式表中使用星号选择器 * 结合属性选择器 right 来选中某些元素。例如：

```css
css
.some-element *right {
  margin-right: 10px;  /* 替换为你需要的样式 */
}
```
这段代码将会选中 .some-element 元素内部所有的右侧元素，并给它们添加一个右边距为10像素的样式。这样做的作用是，解决IE6和IE7中一些特殊情况下元素样式计算错误的问题。
